virttest/libvirt_xml/vm_xml.py

```diff
@@ -1784,28 +1785,130 @@ def __init__(self, virsh_instance=base.virsh):
         self.xml = '<cpu/>'
 
     @staticmethod
-    def marshal_from_cell(item, index, libvirtxml):
+    def marshal_from_cells(item, index, libvirtxml):
         """
-        Convert a dict to cell tag and attributes.
+        Convert an xml object to cache tag and xml element.
         """
-        del index
-        del libvirtxml
-        if not isinstance(item, dict):
-            raise xcepts.LibvirtXMLError("Expected a dictionary of cell "
-                                         "attributes, not a %s"
-                                         % str(item))
-        return ('cell', dict(item))
+        if isinstance(item, NumaCellXML):
+            return 'cell', item
+        else:
+            raise xcepts.LibvirtXMLError("Expected a list of numa cell "
+                                         "instances, not a %s" % str(item))
 
     @staticmethod
-    def marshal_to_cell(tag, attr_dict, index, libvirtxml):
+    def marshal_to_cells(tag, new_treefile, index, libvirtxml):
         """
-        Convert a cell tag and attributes to a dict.
+        Convert a cache tag xml element to an object of CellCacheXML.
         """
-        del index
-        del libvirtxml
         if tag != 'cell':
-            return None
-        return dict(attr_dict)
+            return None     # Don't convert this item
+        newone = NumaCellXML(virsh_instance=libvirtxml.virsh)
+        newone.xmltreefile = new_treefile
+        return newone
+
+
+# Sub-element of cpu/numa
```

Comment:
Please move 1810- 1911 to other place, otherwise the get_feature_list  and following functions of VMCPUXML won't work any more.

Reply:
Absolutely right